pkgs/development/haskell-modules/lib.nix

```diff
@@ -193,6 +193,42 @@ rec {
 
   disableHardening = drv: flags: overrideCabal drv (drv: { hardeningDisable = flags; });
 
+  /* Register the bash completion scripts that are provided by
+   * optparse-applicative. The resulting package will have a
+   * share/bash-completion/completions/${command} file.
+   *
+   *   appendCompletion command pkg
+   *
+   *     command: name of an executable
+   *         pkg: Haskell package that build the executable
+   */
+  appendCompletion = command: pkg:
+    overrideCabal pkg (old: {
+      postInstall = old.postInstall or "" + ''
+        echo "installing bash completion script for ${command}"
```

Comment:
I suppose we'd also want completion files for `zsh` and `fish`? `optparse-applicative` can generate those as well.